multimodal/src/autogluon/multimodal/models/categorical_transformer.py

```diff
@@ -197,7 +203,11 @@ def __init__(
         )
 
         if kv_compression_ratio is not None:
-            n_tokens = self.categorical_feature_tokenizer.n_tokens + 1
+            # number of tokens depends on whether the CLS token is included or not.
+            if self.cls_token:
```

Comment:
It seems that `if self.cls_token` is always True?

Reply:
It is False by default, and should remain False when used together with "fusion_transformer". CLS token is added only once with the fusion transformer.

Reply:
```
self.cls_token = (
            CLSToken(
                d_token=d_token,
                initialization=token_initialization,
            )
            if cls_token
            else nn.Identity()
        )
```
According to this, `self.cls_token` seems always True, whether a nn.Identity() or CLSToken().

Reply:
Yes, I think you are right. Should change to cls_token instead.

Reply:
changed